prime-dotnet-webapi/Controllers/AgreementsController.cs

```diff
@@ -0,0 +1,58 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+
+using Prime.Auth;
+using Prime.Models;
+using Prime.Services;
+using Prime.Models.Api;
+using Prime.ViewModels;
+using Prime.Services.Razor;
+
+namespace Prime.Controllers
+{
+    [Produces("application/json")]
+    [Route("api/[controller]")]
+    [ApiController]
+    public class AgreementsController : ControllerBase
+    {
+        private readonly IAgreementService _agreementService;
+
+        public AgreementsController(IAgreementService agreementService)
+        {
+            _agreementService = agreementService;
+        }
+
+        /// <summary>
+        /// Get a list of the latest enrollee Agreement Versions
+        /// </summary>
+        [HttpGet("enrollee/latest", Name = nameof(GetLatestEnrolleeAgreementVersions))]
+        [Authorize(Roles = Roles.ViewEnrollee)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(typeof(ApiResultResponse<IEnumerable<AgreementVersionViewModel>>), StatusCodes.Status200OK)]
+        public async Task<ActionResult> GetLatestEnrolleeAgreementVersions()
+        {
+            var agreements = await _agreementService.GetLatestEnrolleeAgreementVersionsAsync();
+            return Ok(ApiResponse.Result(agreements));
+        }
+
+        /// /api/agreements/enrollee/2
+        /// <summary>
+        /// Get an enrollee Agreement Version by id
+        /// </summary>
+        /// <param name="agreementId"></param>
+        [HttpGet("enrollee/{agreementId}", Name = nameof(GetEnrolleeAgreementVersionById))]

```

Comment:
This gets the `AgreementVersion` regardless if it's an Enrollee Agreement or not. I feel this URL should just be /api/agreements/{agreementId}